OPAL/tac/src/main/scala/org/opalj/tac/cg/android/AndroidICCAnalysis.scala

```diff
@@ -0,0 +1,1467 @@
+            }
+        }
+        if (addUseSites.nonEmpty) {
+            evaluateUseSites(tacCode, addUseSites.toList, intent)
+        } else implicitIntents += intent
+    }
+
+    def getClassFileFromParameter(p: Expr[DUVar[ValueInformation]], tacCode: TACode[TACMethodParameter, DUVar[ValueInformation]]): Option[ClassFile] = {
+        val defSite = p.asVar.definedBy.head
+        if (defSite > -1) {
+            val statement = tacCode.stmts(defSite)
+            if (statement.isAssignment && statement.asAssignment.expr.isClassConst) {
+                project.classFile(statement.asAssignment.expr.asClassConst.value.asObjectType)
+            } else None
+        } else None
+    }
+
+    /**
+     * Based on the passed name, the parameters are stored in different fields of the intent. If the name belongs to a method,
+     * that makes the intent explicit, an explicit intent is returned.
+     * @param name Name of the analysed method.
+     * @param parameter Parameters of the analysed method.
+     * @param intent The intent the the method is called on.
+     * @param tacCode The TAC code of the method, where the method (name) is called.
+     * @param pc The pc of the call.
+     * @param useSites The use sites of the intent.
+     * @return Returns an explicit intent if an explicit method is called.
+     */
+    def internalFinder(
+        name:      String,
+        parameter: Seq[Expr[DUVar[ValueInformation]]],
+        intent:    ImplicitIntent,
+        tacCode:   TACode[TACMethodParameter, DUVar[ValueInformation]],
+        pc:        Int,
+        useSites:  List[Int]
+    ): Option[ExplicitIntent] = {
+
+        if (explicitMethods.contains(name)) {
+            name match {
+                case `setClass` ⇒
+                    val classFile = getClassFileFromParameter(parameter(1), tacCode)
+                    if (classFile.isDefined) {
+                        val i = new ExplicitIntent(intent.caller, classFile.get, tacCode, useSites)
+                        i.pc = pc
+                        return Some(i);
+                    }
+                case `setClassName` ⇒
+                    val classFile = getClassFileFromParameter(parameter(1), tacCode)
+                    if (classFile.isDefined) {
+                        val i = new ExplicitIntent(intent.caller, classFile.get, tacCode, useSites)
+                        i.pc = pc
+                        return Some(i);
+                    }
+                case `setComponent` ⇒
+                    if (parameter.head.asVar.definedBy.head > -1) {
+                        tacCode.stmts(parameter.head.asVar.definedBy.head).asAssignment.targetVar.usedBy.foreach { u ⇒
+                            if (u > -1) {
+                                val stmt = tacCode.stmts(u)
+                                if (stmt.isNonVirtualMethodCall && stmt.asNonVirtualMethodCall.declaringClass ==
+                                    ObjectType("android/content/ComponentName")) {
```

Comment:
Should probably pre-define that ObjectType (e.g., at the class scope) to avoid its recreation every time this function is called.